connectors/es/utils.py

```diff
@@ -0,0 +1,24 @@
+#
+# Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+# or more contributor license agreements. Licensed under the Elastic License 2.0;
+# you may not use this file except in compliance with the Elastic License 2.0.
+#
+
+
+def extract_index_or_alias(response, name):
```

Comment:
can we add docstrings what s the purpose of this function, it took me a while to digest

Reply:
maybe rename `response` to `get_index_response` just to make clear what we are dealing with here?

Reply:
I've done a full refactoring of this logic instead :)